x-pack/plugins/spaces/server/lib/telemetry_client/telemetry_client.ts

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { ISavedObjectsRepository } from 'src/core/server';
+import { SPACES_TELEMETRY_TYPE } from '../../constants';
+import { CopyOptions, ResolveConflictsOptions } from '../copy_to_spaces/types';
+import { SpacesTelemetry, BooleanCount } from '../../model/spaces_telemetry';
+
+type IncrementCopySavedObjectsOptions = Pick<CopyOptions, 'createNewCopies' | 'overwrite'>;
+type IncrementResolveCopySavedObjectsErrorsOptions = Pick<
+  ResolveConflictsOptions,
+  'createNewCopies'
+>;
+
+const COPY_DEFAULT = Object.freeze({
+  total: 0,
+  createNewCopies: Object.freeze({ enabled: 0, disabled: 0 }),
+  overwrite: Object.freeze({ enabled: 0, disabled: 0 }),
+});
+const RESOLVE_COPY_ERRORS_DEFAULT = Object.freeze({
+  total: 0,
+  createNewCopies: Object.freeze({ enabled: 0, disabled: 0 }),
+});
+
+export class TelemetryClient {
```

Comment:
We need to be very careful about nomenclature here. `TelemetryClient` suggests that this class is a client that provides the Kibana telemetry service but the Kibana Telemetry Service itself is actually provided by the [telemetry plugin](https://github.com/elastic/kibana/tree/master/src/plugins/telemetry#kibana-telemetry-service).   The Telemetry plugin is responsible for collecting all the Kibana usage data and shipping it off to the remote Telemetry service. May I suggest we replace all instances of `Telemetry` in this PR to something else, maybe Usage Data or, at the very least TelemetryData? That would make this class:
```suggestion
export class TelemetryDataClient {
```
or 
```suggestion
export class TelemetryUsageDataClient {
```
It makes the name lengthy but would reduce potential confusion.

Reply:
Thanks for reviewing and letting me know. That makes a lot of sense.

Core already has a CoreUsageDataService that handles usage collection, and it already defines a CoreUsageData type. I thought about renaming that service to something like CoreUsageCollectorService, but I'm hesitant to do so unless we really need to.

Instead, I added 8276465 and opted to rename these all consistently to something else:

Core: 
* CoreTelemetryClient -> CoreUsageStatsClient
* CoreTelemetryService -> CoreUsageStatsService
* CoreTelemetry -> CoreUsageStats

Spaces:
* TelemetryClient -> UsageStatsClient
* TelemetryService -> UsageStatsService
* SpacesTelemetry -> UsageStats

The idea is that Core's CoreUsageData type is a superset of CoreUsageStats, and the Spaces plugin's UsageData type is a superset of UsageStats.

"Usage stats" is information that is collected over time while Kibana is running.
"Usage data" includes usage stats, _plus_ point-in-time info such as configuration, features currently in use, etc.

I'm interested to hear what @elastic/kibana-core thinks about this nomenclature!